agent/engine/engine_integ_test.go

```diff
@@ -937,3 +937,64 @@ func TestSweepContainer(t *testing.T) {
 		t.Error("Expected container to have been sweept but was not")
 	}
 }
+
+// This integ test is meant to validate the docker assumptions related to
+// https://github.com/aws/amazon-ecs-agent/issues/261
+// Namely, this test verifies that Docker does emit a 'die' event after an OOM
+// event if the init dies.  It also verifies that we don't see OOM as fatal
+// unless ther's also a die event.
+func TestOOMHasDieEvent(t *testing.T) {
```

Comment:
I think you can get better verification of the behavior by separating this into two tests:

1. Test `dd` as a child process and let the container continue running (sleep a really long time, infinite loop, whatever).  You'll be able to verify that no `api.ContainerStopped` was emitted and then stop the container manually.
2. Test with `exec`ing `dd` and verify that the `api.ContainerStopped` was emitted with the correct OOM reason.